algorithms/Path/maxflow/tg_maxflow.gsql

```diff
@@ -0,0 +1,186 @@
+CREATE QUERY tg_maxflow(VERTEX source, VERTEX sink, Set<STRING> v_type, SET<STRING> e_type, 
+       SET<STRING> reverse_e_type, STRING cap_attr, STRING cap_type, FLOAT min_flow_threshhold = 0.001, 
+       BOOL print_accum = TRUE, BOOL display_edges = TRUE, BOOL spit_to_file = FALSE, 
+       STRING file_path = "/home/tigergraph/tg_query_output.csv"){ 
+  
+  /*
+   Maxflow algorithm. Finds the maximum amount of flow that source can pushes to sink
+
+   source: start vertex to  every other vertex.     cap_type: capacity data type (UINT, INT,FLOAT,DOUBLE)
+   sink:   end vertex                               min_flow_threshhold: smallest epsilon for flow
+   v_type: vertex types to traverse                 print_accum: print JSON output
+   e_type: edge types to traverse                   display_edges: output edges for visualization
+   reverse_e_type: reverse edge of e_type           spit_to_file: spit output to file
+   cap_attr: attribute for edge capacity            file_path: file to write CSV output to
+     
+  */
+         
+  TYPEDEF TUPLE<INT prev_flow, BOOL is_forward, VERTEX prev> tb_node;
+  GroupByAccum<VERTEX source, VERTEX targ, SumAccum<FLOAT> flow> @@flow_accum;
+  SetAccum<VERTEX> @@curr;
+  SetAccum<EDGE> @@edges;
+  HeapAccum<tb_node>(1, prev_flow DESC) @traceback;
+                              
+  MaxAccum<FLOAT> @@cap_threshold;
+  SumAccum<FLOAT> @@maxflow = 0;
+  MinAccum<FLOAT> @@minflow;
+  OrAccum @is_visited, @@is_found;
+  BOOL minimum_reached = FALSE;
+	FILE f(file_path);
+  @@cap_threshold = min_flow_threshhold;
+         
+  IF cap_type NOT IN ("UINT", "INT", "FLOAT", "DOUBLE") THEN
+		PRINT "wt_type must be UINT, INT, FLOAT, or DOUBLE" AS errMsg;
+		RETURN;
+	END;    
+  
+  ##### Initialize #####
+  init = {v_type};       
+  init = SELECT s
+         FROM init:s - (e_type:e) -> v_type:t
+         ACCUM 
+          FLOAT fl = 0,
+          CASE cap_type
+            WHEN "UINT" THEN
+					    fl = e.getAttr(cap_attr, "UINT")
+            WHEN "INT" THEN
+					    fl = e.getAttr(cap_attr, "INT")
+            WHEN "FLOAT" THEN
+              fl = e.getAttr(cap_attr, "FLOAT")
+            WHEN "DOUBLE" THEN
+              fl = e.getAttr(cap_attr, "DOUBLE")
+          END,
+          @@flow_accum += (s, t -> 0),
+          IF s == source THEN @@cap_threshold += fl END;
+
+  @@cap_threshold = pow(3, float_to_int(log(@@cap_threshold)/log(3)));
```

Comment:
add some comments on how cap_threshold is calculated/ used?
a reference to the capacity scaling algorithm will be helpful
you can do it as general information comments at the beginning or when the cap_threshold is used